tsl/src/fdw/deparse.c

```diff
@@ -884,20 +1047,25 @@ deparseSelectSql(List *tlist, bool is_subquery, List **retrieved_attrs, deparse_
 		 * required to be fetched from the data node.
 		 */
 		RangeTblEntry *rte = planner_rt_fetch(foreignrel->relid, root);
+		List *dc_varattnos = NIL;
 
 		/*
 		 * Core code already has some lock on each rel being planned, so we
 		 * can use NoLock here.
 		 */
 		Relation rel = table_open(rte->relid, NoLock);
 
+		if (root->parse->distinctClause != NIL)
+			deparseDistinctClause(buf, &dc_varattnos, context);
```

Comment:
```suggestion
			dc_varattnos = deparseDistinctClause(buf, &dc_varattnos, context);
```

Reply:
Again, here was trying to be similar in style to `deparseSelectStmtForRel` and `deparseSelectSql` which get the list as an argument instead of a return value. Stylistic difference..